File: server/src/main/java/com/logiflow/server/models/Order.java
package com.logiflow.server.models;

import jakarta.persistence.*;
import lombok.Data;
import java.math.BigDecimal;
import java.time.LocalDateTime;

import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;

@Data
@Entity
@Table(name = "orders")
public class Order {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "order_id")
    private Integer orderId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "trip_id")
    private Trip trip;

    @Column(name = "customer_name", length = 100, nullable = false)
    private String customerName;

    @Column(name = "customer_phone", length = 20, nullable = true)
    private String customerPhone;

    @Column(name = "pickup_address", length = 255, nullable = false)
    private String pickupAddress;

    @Column(name = "pickup_type", length = 20, nullable = true)
    @Enumerated(EnumType.STRING)
    private PickupType pickupType;

    @Column(name = "container_number", length = 50, nullable = true)
    private String containerNumber;

    @Column(name = "terminal_name", length = 100, nullable = true)
    private String terminalName;

    @Column(name = "warehouse_name", length = 100, nullable = true)
    private String warehouseName;

    @Column(name = "dock_number", length = 50, nullable = true)
    private String dockNumber;

    @Column(name = "delivery_address", length = 255, nullable = false)
    private String deliveryAddress;

    @Column(name = "package_details", length = 500, nullable = true)
    private String packageDetails;



    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "customer_id")
    private User customer;
    @Column(name = "distance_km", precision = 10, scale = 2, nullable = true)
    private BigDecimal distanceKm;

    @Column(name = "weight_kg", precision = 10, scale = 2, nullable = true)
    private BigDecimal weightKg;

    @Column(name = "package_value", precision = 15, scale = 2, nullable = true)
    private BigDecimal packageValue;

    @Column(name = "shipping_fee", precision = 15, scale = 2, nullable = true)
    private BigDecimal shippingFee;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "created_by")
    private User createdBy;

    @Column(name = "priority_level", nullable = false)
    @Enumerated(EnumType.STRING)
    private PriorityLevel priorityLevel = PriorityLevel.NORMAL;

    @Column(name = "order_status", nullable = false)
    @Enumerated(EnumType.STRING)
    private OrderStatus orderStatus = OrderStatus.PENDING;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public static enum PriorityLevel {
        NORMAL, URGENT
    }

    public static enum PickupType {
        PORT, WAREHOUSE
    }

    public static enum OrderStatus {
        PENDING, ASSIGNED, IN_TRANSIT, DELIVERED, CANCELLED
    }
}

File: client_mobile/lib/models/driver/order.dart
class DriverOrder {
  final int orderId;
  final String customerName;
  final String? customerPhone;
  final String pickupAddress;
  final String? pickupType;
  final String? containerNumber;
  final String? terminalName;
  final String? warehouseName;
  final String? dockNumber;
  final String deliveryAddress;
  final double? pickupLat;
  final double? pickupLng;
  final double? deliveryLat;
  final double? deliveryLng;
  final String? packageDetails;
  final double? weightTons;
  final double? packageValue;
  final double? distanceKm;
  final String? status;
  final String? orderStatus;
  final String? priority;
  final String? priorityLevel;
  final String? delayReason;

  const DriverOrder({
    required this.orderId,
    required this.customerName,
    this.customerPhone,
    required this.pickupAddress,
    this.pickupType,
    this.containerNumber,
    this.terminalName,
    this.warehouseName,
    this.dockNumber,
    required this.deliveryAddress,
    this.pickupLat,
    this.pickupLng,
    this.deliveryLat,
    this.deliveryLng,
    this.packageDetails,
    this.weightTons,
    this.packageValue,
    this.distanceKm,
    this.status,
    this.orderStatus,
    this.priority,
    this.priorityLevel,
    this.delayReason,
  });

  factory DriverOrder.fromJson(Map<String, dynamic> json) {
    return DriverOrder(
      orderId: (json['orderId'] as num).toInt(),
      customerName: json['customerName'],
      customerPhone: json['customerPhone'],
      pickupAddress: json['pickupAddress'],
      pickupType: json['pickupType'],
      containerNumber: json['containerNumber'],
      terminalName: json['terminalName'],
      warehouseName: json['warehouseName'],
      dockNumber: json['dockNumber'],
      deliveryAddress: json['deliveryAddress'],
      pickupLat: json['pickupLat']?.toDouble(),
      pickupLng: json['pickupLng']?.toDouble(),
      deliveryLat: json['deliveryLat']?.toDouble(),
      deliveryLng: json['deliveryLng']?.toDouble(),
      packageDetails: json['packageDetails'],
      weightTons: json['weightTons']?.toDouble(),
      packageValue: json['packageValue']?.toDouble(),
      distanceKm: json['distanceKm']?.toDouble(),
      status: json['status'],
      orderStatus: json['orderStatus'],
      priority: json['priority'],
      priorityLevel: json['priorityLevel'],
      delayReason: json['delayReason'],
    );
  }

  Map<String, dynamic> toJson() {
    return {
      'orderId': orderId,
      'customerName': customerName,
      'customerPhone': customerPhone,
      'pickupAddress': pickupAddress,
      'pickupType': pickupType,
      'containerNumber': containerNumber,
      'terminalName': terminalName,
      'warehouseName': warehouseName,
      'dockNumber': dockNumber,
      'deliveryAddress': deliveryAddress,
      'pickupLat': pickupLat,
      'pickupLng': pickupLng,
      'deliveryLat': deliveryLat,
      'deliveryLng': deliveryLng,
      'packageDetails': packageDetails,
      'weightTons': weightTons,
      'packageValue': packageValue,
      'distanceKm': distanceKm,
      'status': status,
      'orderStatus': orderStatus,
      'priority': priority,
      'priorityLevel': priorityLevel,
      'delayReason': delayReason,
    };
  }
}
